### ISSUE TITLE
Two column mode page breaks only break columns

### ISSUE THREAD

**Issue by jonnyl2** [koreader/koreader#11124]
* KOReader version: 2023.10-18
* Device: Android and Kobo

#### Issue

Page breaks in 2-column mode only start a new column, not necessarily a new page.

In addition to new chapters starting in the right-hand column not looking very pleasing to the eye (in my opinion), it also makes marking those pages as 'user hidden flows' less precise.

#### Steps to reproduce

Open any reflowable book with page breaks and set to Two column mode. 

Or, set page breaks in style tweaks. 

**Example**:

Before style tweak page break at \<H2>:

![Reader_2023-11-18_094104](https://github.com/koreader/koreader/assets/95502269/1933dbb5-288c-401e-b458-167451bb3af2)

After \<H2> page breaks:

![Reader_2023-11-18_094129](https://github.com/koreader/koreader/assets/95502269/774b7eba-ec6e-4852-a852-fa657fd3f5e9)

**Comment by Frenzie** [comment]
> Page breaks in 2-column mode only start a new column, not necessarily a new page.

Sssh, don't pull back the magic behind how it works. ;-P

But @poire-z might be able to say if it can be resolved with relative ease or if it'd be incredibly difficult.

**Comment by poire-z** [comment]
Depends on how you think about these "2 columns". For me (may be biased by knowing how it's implemented), your 2nd screenshot looks like how I would expect it: 2 pages on a screen, a chapter starting on the right page, as in paper books :). But I get that it's random, and we both get 50% chances at each chapter of being happy.

Anyway, technically, crengine does all that. There's really no notion of "2 columns" for it: it renders pages on a target width of half the screen, and put 2 of them on the final drawn bitmap. It is us who named that feature in frontend "2 columns", just because we wanted to have incremental page numbers (long ago, page numbers were incremented by 2 on each page turn).

Also technically, crengine first renders the content on this target width as content flows, on a big papyrus roll; it applies styles during this process. The page splitting process happens at the end, and cut that roll, way later after CSS is applied.
I know there are CSS (or EPUB?) stuff to handle column and start things on odd or even pages. https://developer.mozilla.org/en-US/docs/Web/CSS/break-before may be, dunno if there's some other stuff.
It would mean passing some flags from the first phase down to the late phase of page splitting, so that late phase could see this new page must be even/odd/left/right/recto/verso, and emit a blank page if needed. Which might be possible, but the interest is a bit low to me tbh (but adding that on my tothink list).

Anyway, given the various expectations explicited above :) it should probably be best left to style tweaks and handling existing CSS properties, so users can tweak them to their taste.

**Comment by Frenzie** [comment]
> Depends on how you think about these "2 columns". For me (may be biased by knowing how it's implemented), your 2nd screenshot looks like how I would expect it: 2 pages on a screen, a chapter starting on the right page, as in paper books :). 

Yup! It's actually great. Though strictly there's page-break-before: left/right for that, while our tweaks use always iirc.

**Comment by jonnyl2** [comment]
>Depends on how you think about these "2 columns". For me (may be biased by knowing how it's implemented), your 2nd screenshot looks like how I would expect it: 2 pages on a screen, a chapter starting on the right page, as in paper books :). 

I can see that point of view as well, especially in landscape orientation. Agreed that a user-setting/tweak might be the best solution.
